packages/hub/src/graphql/queries/runSquiggle.ts

```diff
@@ -1,14 +1,25 @@
 import { Prisma } from "@prisma/client";
 import crypto from "crypto";
 
-import { SqProject, SqValue } from "@quri/squiggle-lang";
-
-import { DEFAULT_SEED, SAMPLE_COUNT_DEFAULT } from "@/constants";
+import {
+  SqLinker,
+  SqOutputResult,
+  SqProject,
+  SqValue,
+} from "@quri/squiggle-lang";
+
+import { SAMPLE_COUNT_DEFAULT, XY_POINT_LENGTH_DEFAULT } from "@/constants";
 import { builder } from "@/graphql/builder";
 import { prisma } from "@/prisma";
+import { parseSourceId } from "@/squiggle/components/linker";
+
+import { NotFoundError } from "../errors/NotFoundError";
 
-function getKey(code: string): string {
-  return crypto.createHash("md5").update(code).digest("base64");
+function getKey(code: string, seed: string): string {
+  return crypto
+    .createHash("md5")
+    .update(`__EXPORT__:${code}, __SEED__: ${seed}`)
```

Comment:
I wanted to make something that seemed very unlikely to accidentally be hit. I assume it's possible for someone to name the end of their code __SEED__ and mess this up, but it seems unlikely. We could also use a squiggle-invalid code here, if we're extra paranoid. I think though that we can kind of overlook this now, as it's hacky now anyway and needs to be rethought.

Reply:
Thoughts:
- crypto hash is good (similar to git)
- no reason to use MD5 in 2024, go with SHA-256 by default
- I'd probably do `SHA-256(SHA-256(code), seed)`, or `SHA-256(JSON.serialize({ code, seed }))`, but I agree it doesn't matter now, we can refill all caches later

Reply:
Most of this code was there before, but I can make simple updates.